cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/CreateSchedulesStage.java

```diff
@@ -81,54 +76,39 @@ public void process(ApplicationWithPrograms input) throws Exception {
         // hence it is useless to update the schedule
         continue;
       }
-      ProgramType programType = ProgramType.valueOfSchedulableType(newScheduleSpec.getProgram().getProgramType());
-      ProgramId programId = appId.program(programType, newScheduleSpec.getProgram().getProgramName());
 
       // if the schedule differ in schedule type then we have deleted the existing one earlier in DeleteScheduleStage.
       // create it with the new type and spec here. See CDAP-8918 for details.
```

Comment:
Without the REST API changes, there are still some schedules only exist in AppSpec and old scheduler but not in ProgramSchedule store. We still need check app spec when deleting schedules